src/main/groovy/org/pih/warehouse/core/HistoryItem.groovy

```diff
@@ -0,0 +1,15 @@
+package org.pih.warehouse.core
+
+class HistoryItem<T> {
+    String identifier
+
+    Date date
+
+    // FIXME: I didn't know what name suits the best for the location
+    Location associatedLocation
+
+    T parentObject
+
+    // FIXME: Introduced this property to be able to determine which history item is pointing to e.g. received event, which one to shipped event etc.
+    EventCode eventCode
```

Comment:
So the idea is that this HistoryItem is generic enough to be applied to all features that need history? What if we need the history item to contain some feature-specific data?

Maybe it'd be more flexible if each feature was forced to implement their own ShipmentHistoryItem which extends HistoryItem.

Or maybe that's not needed and I'm overthinking it. I suppose my question is what does the future of this class look like?